pkg/types/jar/v0.0.1/entry.go

```diff
@@ -0,0 +1,342 @@
+/*
+Copyright © 2021 The Rekor Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package jar
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"path"
+	"reflect"
+	"strings"
+
+	"github.com/sigstore/rekor/pkg/log"
+	"github.com/sigstore/rekor/pkg/pki"
+	"github.com/sigstore/rekor/pkg/types"
+	"github.com/sigstore/rekor/pkg/types/jar"
+	"github.com/sigstore/rekor/pkg/util"
+
+	"github.com/asaskevich/govalidator"
+
+	"github.com/go-openapi/strfmt"
+
+	"github.com/go-openapi/swag"
+	"github.com/mitchellh/mapstructure"
+	jarutils "github.com/sassoftware/relic/lib/signjar"
+	"github.com/sigstore/rekor/pkg/generated/models"
+)
+
+const (
+	APIVERSION = "0.0.1"
+)
+
+func init() {
+	jar.SemVerToFacFnMap.Set(APIVERSION, NewEntry)
+}
+
+type V001Entry struct {
+	JARModel                models.JarV001Schema
+	fetchedExternalEntities bool
+	jarObj                  *jarutils.JarSignature
+	keyObj                  pki.PublicKey
+	sigObj                  pki.Signature
+}
+
+func (v V001Entry) APIVersion() string {
+	return APIVERSION
+}
+
+func NewEntry() types.EntryImpl {
+	return &V001Entry{}
+}
+
+func Base64StringtoByteArray() mapstructure.DecodeHookFunc {
```

Comment:
we have this defined in two places already, should we put it somewhere common?

Reply:
yes, this and the semver stuff could be abstracted out. I will open an issue for this.